src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1105,15 +1105,14 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         const queryRunner = this.obtainQueryRunner();
         let transactionStartedByUs: boolean = false;
         try {
-
             // start transaction if it was enabled
             if (this.expressionMap.useTransaction === true && queryRunner.isTransactionActive === false) {
                 await queryRunner.startTransaction();
                 transactionStartedByUs = true;
             }
 
             this.expressionMap.queryEntity = true;
-            this.applyFindOptions();
+
```

Comment:
how do I understand motivation behind this change?

Reply:
I think we should skip changes related to find options. Both approach looks like they're doing exactly same thing, but in a different way. 
As for this change it's probably because of this line https://github.com/Benjamin-Dobell/typeorm/blob/86134f7ab9e9cd07d33ac360d928236df77dd95c/src/query-builder/SelectQueryBuilder.ts#L1845 - we don't have metadata available in some contexts(migrations) and we're using getter witch throws an error instead of `hasMetadata` method(in current version of next `hasMetadata` is used).

Reply:
I think we should skip changes related to findOptions in QueryBuilder. Both approaches looks like they're doing same thing with a different way(apart from find options precedence overs query builder methods).
As for the change - it's probably because we don't have metadata available in all contexts(in migrations), and there is a check on `metadata` property(which throws an error if there is none) instead of `hasMetadata` method. Current version of next uses `hasMetadata` in code where similar problem happened.

Reply:
@pleerock To understand this change you should be looking at it in the context of the commit, which is why this PR is split up into several commits:

https://github.com/typeorm/typeorm/pull/5336/commits/aa023e47cd1437ea3be793481eb14f93b3437468

The commit message states that I've fixed lock mode. You'll then note that I've not simply deleted this line, but rather moved all calls to `applyFindOptions` to be _before_ any usages of `this.expressionMap.lockMode`.

Prior to this commit we were incorrectly checking the lock mode before the find options had been applied, thus it was effectively being ignored when the query builder was being used as part of a `find`.